BUILD.md

```diff
@@ -16,43 +17,61 @@
 
 ### <a id="install_header_only_cpp"></a>Header-only C++
 
-RAFT depends on many different core libraries such as `thrust`, `cub`, `cucollections`, and `rmm`, which will be downloaded automatically by `cmake` even when only installing the headers. It's important to note that while all the headers will be installed and available, some parts of the RAFT API depend on libraries like `FAISS`, which can also be downloaded in the RAFT build but will need to be told to do so.
+`build.sh` uses [rapids-cmake](https://github.com/rapidsai/rapids-cmake), which will automatically download any dependencies which are not already installed. It's important to note that while all the headers will be installed and available, some parts of the RAFT API depend on libraries like `FAISS`, which will need to be explicitly enabled in `build.sh`.
 
-The following example builds and installs raft in header-only mode:
+The following example will download the needed dependencies and install the RAFT headers into `$INSTALL_PREFIX/include/raft`. The `--install` flag can be omitted to just have the build download the needed dependencies. Since RAFT is primarily used at build-time, the dependencies will never be installed by the RAFT build, with the exception of building FAISS statically into the shared libraries.
 ```bash
-./build.sh libraft
+./build.sh libraft --install
 ```
 
 ###<a id="shared_cpp_libs"></a>C++ Shared Libraries (optional)
 
-Shared libraries are provided to speed up compile times for larger libraries which may heavily utilize some of the APIs. These shared libraries can also significantly improve re-compile times while developing against the APIs.
-
-Build all the shared libraries by passing `--compile-libs` flag to `build.sh`:
-
+For larger projects which make heavy use of the pairwise distances or nearest neighbors APIs, shared libraries can be built to speed up compile times. These shared libraries can also significantly improve re-compile times both while developing RAFT and developing against the APIs. Build all of the available shared libraries by passing `--compile-libs` flag to `build.sh`:
 ```bash
 ./build.sh libraft --compile-libs
 ```
 
-To remain flexible, the individual shared libraries have their own flags and multiple can be used (though currently only the `nn` and `distance` packages contain shared libraries):
+Individual shared libraries have their own flags and multiple can be used (though currently only the `nn` and `distance` packages contain shared libraries):
 ```bash
 ./build.sh libraft --compile-nn --compile-dist
 ```
 
-###<a id="gtests"></a>Googletests
+Add the `--install` flag to the above example to also install the shared libraries into `$INSTALL_PREFIX/lib`.
+
+###<a id="gtests"></a>Tests
```

Comment:
```suggestion
### <a id="gtests"></a>Tests
```
(same for other section headers missing a space since I think they're not rendering right)

Reply:
The auto generated section titles do some weird things with different characters which was non-obvious. I prefer the html also for maintainability because it makes very explicit the link which corresponds to the title.